src/main/java/org/llorllale/cactoos/matchers/HasEntry.java

```diff
@@ -49,7 +49,7 @@
  * @param <V> Type of value.
  * @since 1.0.0
  */
-public final class HasEntry<K, V> extends MatcherEnvelope<Map<K, V>> {
+public final class HasEntry<K, V> extends MatcherEnvelope<Map<? extends K, ? extends V>> {
```

Comment:
@andreoss this is a bit tricky and shows the limits of Java concerning variance. Basically, if we wanted to use the Map type as a function from a key to a value (i.e., using `theMap.get(o)` in the code of `HasEntry`), then it would make sense to do as you propose. But since the implementation of `HasEntry` is about applying two matchers, one to the key and one to the value to match an entry, then we are manipulating the Map just as a collection of entries, and thus normal variance applies with `extends` here. Same with `IsEntry` below.

In a perfect world where variance is correctly supported (e.g., Scala), the variance (`? extends` or `? super`) would not be declared here, at the usage site, but directly in the Map interface, per method for example and there would be no problem ^^ Unfortunately here we have to take into account what we are expecting of `Map`, and it's to be a collection of entries.